tests/experiment/test_results.py

```diff
@@ -36,6 +38,9 @@
 from data.procedure_for_testing import RandomProcedure
 
 
+u = pint.get_application_registry()
```

Comment:
I got this usage pattern in my bones to not duplicate uregs, so reuse the existing one from pymeasure.experiment.results instead. Is that not needed anymore?

Reply:
get_application_registry does exactly what you want: If the application has already a registry, it returns it, such that you use that already existing registry again. If no registry exists, it creates one.
With that method you do not have to remember, where you defined it in the first place.

Reply:
I checked it, and get_application_registry works that way only after having called set_application_registry somewhere else.
So we have to declare it at some single point of entry (later on).

Reply:
From this [documentation section](https://pint.readthedocs.io/en/stable/tutorial.html#using-pint-in-your-projects) and [this comment](https://github.com/hgrecco/pint/issues/1273#issuecomment-805369430) I take away that the best way is to put this into our codebase somewhere and use `ureg` it from there
```
import pint
ureg = pint.get_application_registry()
```
(we maybe need to put it into a separate moduel to avoid circular imports, but I haven't confirmed that)

cf also [this issue](https://github.com/hgrecco/pint/issues/1365).

Reply:
pint 0.19 requires python>=3.8, but pymeasure still supports python3.7

Reply:
> (we maybe need to put it into a separate moduel to avoid circular imports, but I haven't confirmed that)

If there is not an application_registry (via set_application_registry), it will create one. as we want to keep the registry a user thing. I propose to move (later on) `ureg = pint.get_application_registry()` in the `pymeasure.__init__.py`, such that all pymeasure modules use this registry, which in turn gets the application registry set by the user (if any).